clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java

```diff
@@ -186,6 +187,19 @@ public void partitionAssignmentChangeOnPatternSubscription() {
         assertTrue(state.assignedPartitions().isEmpty());
         assertEquals(0, state.numAssignedPartitions());
     }
+    @Test
+    public void testSubscriptionPatternInclusionInSubscriptionState() {
+        SubscriptionPattern pattern = new SubscriptionPattern("*t");
+
+        state.subscribe(pattern, Optional.of(rebalanceListener));
+        assertTrue(state.assignedPartitions().isEmpty());
+        assertEquals(0, state.numAssignedPartitions());
+        assertTrue(state.subscriptionPattern().equals(pattern));
```

Comment:
You also need to verify that the java pattern is not set. In the tests that verify that the java pattern is set you need to verify the subscription pattern is not set.